publication-commons/src/main/java/no/unit/nva/publication/model/business/Resource.java

```diff
@@ -219,15 +225,55 @@ public Resource fetch(ResourceService resourceService) throws NotFoundException
         return resourceService.getResourceByIdentifier(this.getIdentifier());
     }
 
-    public void republish(ResourceService resourceService, UserInstance userInstance) {
-        if (!UNPUBLISHED.equals(getStatus())) {
-            throw new IllegalStateException("Only unpublished resource can be republished");
+    public void publish(ResourceService resourceService, UserInstance userInstance) throws NotFoundException {
+        var resource = this.fetch(resourceService);
+        if (resource.isNotPublished()) {
+            resource.publish(userInstance);
+            resourceService.updateResource(resource);
+        }
+    }
+
+    private void publish(UserInstance userInstance) {
+        if (this.isPublished()) {
+            return;
+        }
+        if (this.isNotPublishable()) {
+            throw new IllegalStateException("Resource is not publishable!");
+        } else {
+            this.setStatus(PUBLISHED);
+            var currentTime = Instant.now();
+            this.setPublishedDate(currentTime);
+            this.setResourceEvent(PublishedResourceEvent.create(userInstance, currentTime));
+        }
+    }
+
+    private boolean isNotPublishable() {
+        return !PUBLISHABLE_STATUSES.contains(this.getStatus())
+               || Optional.ofNullable(this.getEntityDescription()).map(EntityDescription::getMainTitle).isEmpty();
+    }
+
+    private boolean isNotPublished() {
+        return !isPublished();
+    }
+
+    private boolean isPublished() {
+        return PUBLISHED.equals(this.getStatus());
+    }
+
+    public void republish(ResourceService resourceService, UserInstance userInstance) throws NotFoundException {
+        var resource = this.fetch(resourceService);
+        resource.republish(userInstance);
+        resourceService.updateResource(resource);
+    }
+
+    private void republish(UserInstance userInstance) {
```

Comment:
nitpick: Could `republish` call `publish` directly, after checking if status is valid? The rest of the code appears to be duplicate.

Reply:
We could do it, but we set to different resource events for these two actions, so I think it is better to differentiate:
When publishing:
`this.setResourceEvent(PublishedResourceEvent.create(userInstance, currentTime));`
When republishing:
`this.setResourceEvent(RepublishedResourceEvent.create(userInstance, timestamp));
`